daemon/networkdriver/bridge/driver.go

```diff
@@ -289,6 +307,20 @@ func InitDriver(config *Config) error {
 			return err
 		}
 		globalIPv6Network = subnet
+
+		if config.DefaultGatewayIPv6 != "" {
+			gatewayIPv6 = net.ParseIP(config.DefaultGatewayIPv6)
+
+			if gatewayIPv6 == nil {
+				return fmt.Errorf("Bad parameter: invalid gateway ip %s", config.DefaultGatewayIPv6)
+			}
+
+			if !globalIPv6Network.Contains(gatewayIPv6) {
+				return fmt.Errorf("Gateway ip %s must be part of the network %s", config.DefaultGatewayIPv6, globalIPv6Network.String())
+			}
+
+			ipAllocator.RequestIP(globalIPv6Network, gatewayIPv6)
+		}
```

Comment:
Because they only manipulate `net.IP` and `net.IPNet`, I believe this code block (covering IPv6) and the one above (covering IPv4) can entirely be factored out in a function.

Not a blocker for merge considering that there's [a death timer](https://github.com/docker/libnetwork) on this file anyway.